cmd/serf/command/agent/command.go

```diff
@@ -176,6 +179,19 @@ func (c *Command) readConfig() *Config {
 		return nil
 	}
 
+	if config.MDNS.Interface != "" {
```

Comment:
We should also handle the error case where `mdns.interface` is specified w/o `discovery`.

Reply:
I added a check but I was not sure whether we should also add a check for the disable-ip-v(4|6) flags

Reply:
I don't think we need it, because it is valid to inherit the top-level `iface` in that case. 👍